### PR TITLE
Update sample for TypeSpec compiler v0.43.0

### PR DESCRIPTION
**Pull request by mikeharder**
I think this is ready to merge:

* The failure in TypeSpecValidation is tracked by https://github.com/Azure/azure-sdk-tools/issues/6064
* I believe check "SDK azure-sdk-for-net-track2" is known to be failing consistently on TypeSpec PRs

### CODE REVIEW

**Comment by raych1** [comment]
@mikeharder , is `TypeSpecValidation` failure expected? When will this issue be fixed?

**Comment by mikeharder** [comment]
> @mikeharder , is `TypeSpecValidation` failure expected? When will this issue be fixed?

This failure was just discovered today, and is tracked by https://github.com/Azure/azure-sdk-tools/issues/6064.  I will work with @ckairen to get this fixed, probably by EOD Friday.

File: specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.tsp [hunk 1]
```diff
@@ -4,5 +4,5 @@ model FakedSharedModel {
   tag: string;
 
   @doc("The created date.")
-  createdDate: zonedDateTime;
-}
\ No newline at end of file
+  createdDate: utcDateTime;
```
Comment (JeffreyRichter) [inline specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.tsp:7]:
```suggestion
  createdAt: utcDateTime;
```

To follow our naming guidelines: https://github.com/microsoft/api-guidelines/blob/vNext/azure/ConsiderationsForServiceDesign.md#common-names

**Review by JeffreyRichter — APPROVED** [review]

File: specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.tsp (hunk 1 above)
Comment (mikeharder) [reply specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.tsp:7]: @weshaggard: Do you think this property name should be updated in the sample per @JeffreyRichter suggestion?  If so, I will update this in a separate PR.

**Comment by mikeharder** [comment]
`TypeSpecValidation` is now "passing", but it's skipping our project instead of actually running validation, so there is still a problem:

```
swaggersFromTypeSpec: ["specification/contosowidgetmanager/data-plane/Azure.Contoso.WidgetManager/preview/2022-11-01-preview/widgets.json"]
changedTypeSpecPackages: ["specification/contosowidgetmanager/Contoso.WidgetManager.Shared","specification/contosowidgetmanager/Contoso.WidgetManager"]
typespecPackages: ["specification/contosowidgetmanager/Contoso.WidgetManager"]
Validating Package: specification/contosowidgetmanager/Contoso.WidgetManager
Skipping Package: specification/contosowidgetmanager/Contoso.WidgetManager
```

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2746379&view=logs&j=33174e5f-e491-5dfc-305d-c09df824202b&t=0f377f82-3711-5097-078f-1ce87e974d2b&l=21

**Comment by mikeharder** [comment]
> `TypeSpecValidation` is now "passing", but it's skipping our project instead of actually running validation, so there is still a problem:
> 
> ```
> swaggersFromTypeSpec: ["specification/contosowidgetmanager/data-plane/Azure.Contoso.WidgetManager/preview/2022-11-01-preview/widgets.json"]
> changedTypeSpecPackages: ["specification/contosowidgetmanager/Contoso.WidgetManager.Shared","specification/contosowidgetmanager/Contoso.WidgetManager"]
> typespecPackages: ["specification/contosowidgetmanager/Contoso.WidgetManager"]
> Validating Package: specification/contosowidgetmanager/Contoso.WidgetManager
> Skipping Package: specification/contosowidgetmanager/Contoso.WidgetManager
> ```
> 
> https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2746379&view=logs&j=33174e5f-e491-5dfc-305d-c09df824202b&t=0f377f82-3711-5097-078f-1ce87e974d2b&l=21

Fix for this should be deployed to prod tomorrow

File: specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.tsp (hunk 1 above)
Comment (weshaggard) [reply specification/contosowidgetmanager/Contoso.WidgetManager.Shared/main.tsp:7]: Yes I'd like to keep the sample matching the guidelines.

**Pull request merged**